infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LocationTrackService.kt

```diff
@@ -364,23 +364,26 @@ class LocationTrackService(
     )?.let(switchDao::fetch)?.let { switch -> LayoutSwitchIdAndName(id, switch.name) }
 
     @Transactional
-    fun updateTopology(
+    fun calculateLocationTrackTopology(
```

Comment:
Tuo `update` termi on/oli kyllä hiukan hähmeä: Välillä sitä käytetään funktioissa, jotka päivittävät muistinvaraisesti tietorakennetta, ja toisaalla taas sellaisissa, jotka päivittävät myös kantaa. En osaa sanoa, että onko tämä sen nimitys paljoa parempi.

Tämähän funktio ei siis aiemmin tehnyt eikä tee tämänkään jälkeen kantaan muutoksia, vaikka (saattaakin) sieltä lueskella.

Yleisempänä huomiona saattaisi olla jatkossa järkevää pyrkiä funktioida ns. lasku- ja kantaankirjoitusoperaatiot erillisiksi.

Reply:
Eriyttäminen olisi kyllä hyvä. Itse haluaisin ehkä vielä viedä tämän askeleen pidemmälle ja eriyttää kantaluvutkin mahdollisimman hyvin varsinaisesta laskennasta (mahdollisuuksien mukaan ensin haetaan kannasta tarvittavat jutut, sitten lasketaan erillisenä.) Tällöin esim. tuon laskennan pystyy yksikkötestaamaan ilman kantayhteyttä, ja toisaalta mikään ulkoisesti laskuoperaation näköinen juttu ei voi failata siihen että esim. kantayhteys kuolee alta.